shared/api/certificate.go

```diff
@@ -72,7 +72,31 @@ type CertificatePut struct {
 //
 // swagger:model
 type Certificate struct {
-	CertificatePut `yaml:",inline"`
+	// Name associated with the certificate
+	// Example: castiana
+	Name string `json:"name" yaml:"name"`
+
+	// Usage type for the certificate
+	// Example: client
+	Type string `json:"type" yaml:"type"`
+
+	// Whether to limit the certificate to listed projects
+	// Example: true
+	//
+	// API extension: certificate_project
+	Restricted bool `json:"restricted" yaml:"restricted"`
+
+	// List of allowed projects (applies when restricted)
+	// Example: ["default", "foo", "bar"]
+	//
+	// API extension: certificate_project
+	Projects []string `json:"projects" yaml:"projects"`
+
+	// The certificate itself, as PEM encoded X509 (or as base64 encoded X509 on POST)
```

Comment:
If we're unembedding these API types, now might be a good time to also unembed `CertificatePut` from `CertificatesPost` so that the base64 encoding of the certificate on POST can be better reflected in the API docs.

Reply:
Sounds good

Reply:
Very good point. Will address that.

Reply:
@MusicDin this comment doesnt seem appropriate here now.

Reply:
Thanks. Fixed.